app/src/main/java/com/example/bookwormadventuresdeluxe2/AddOrEditBooksActivity.java

```diff
@@ -118,6 +153,51 @@ public void uploadPhoto(View view)
         startActivityForResult(Intent.createChooser(uploadPhotoIntent, getResources().getString(R.string.select_photo)), REQUEST_IMAGE_UPLOAD);
     }
 
+    /**
+     * Deletes the picture associated with the book being edited
+     */
+    private void deletePhoto()
+    {
+        String imageUrl;
+        if (editingBook)
+        {
+            imageUrl = this.bookToEdit.getImageUrl();
+            // remove association of book to the image url
+            this.bookToEdit.setImageUrl("");
+        }
+        else
+        {
+            /* This will occur when a user uploads a picture
+             * when adding a new book then deletes it before saving the new book*/
+            imageUrl = this.bookPhotoDowloadUrl;
+        }
+
+        /* This should never be called when the book does not have an image*/
+        if (imageUrl.equals(""))
+        {
+            throw new IllegalStateException("deletePhoto should not have been called!");
+        }
+
+        // Delete image from firebase
+        FirebaseStorage storage = FirebaseStorage.getInstance();
```

Comment:
Since we do this twice in AddOrEditBooksActivity I think we should extract it to a function